### PR TITLE
Kill the core API package.

### PR DESCRIPTION
**Pull request by ben-clayton**

### CODE REVIEW

File: gapis/gfxapi/gles/replay.go [hunk 1]
```diff
@@ -261,84 +261,88 @@ func (t *destroyResourcesAtEOS) Transform(ctx context.Context, id atom.ID, a ato
 func (t *destroyResourcesAtEOS) Flush(ctx context.Context, out transform.Writer) {
 	id := atom.NoID
 	s := out.State()
-	thread := (uint64)(GetState(s).CurrentThread)
-	c := GetState(s).GetContext(thread)
-	if c == nil {
-		return
-	}
+	for t, c := range GetState(s).Contexts {
```
Comment (dsrbecky) [inline gapis/gfxapi/gles/replay.go]: This still does not necessarily free all contexts right?
Let's just create an issue to track this? (or TODO)

File: gapis/gfxapi/gles/replay.go [hunk 2]
```diff
@@ -261,84 +261,88 @@ func (t *destroyResourcesAtEOS) Transform(ctx context.Context, id atom.ID, a ato
...
 
-	// Delete all Renderbuffers.
-	renderbuffers := make([]RenderbufferId, 0, len(c.Objects.Shared.Renderbuffers)-3)
-	for renderbufferId := range c.Objects.Shared.Renderbuffers {
-		// Skip virtual renderbuffers: backbuffer_color(-1), backbuffer_depth(-2), backbuffer_stencil(-3).
-		if renderbufferId < 0xf0000000 {
-			renderbuffers = append(renderbuffers, renderbufferId)
+		// TODO: This is just looping over the threads - we need to loop over
+		// all contexts and destroy everything there. That requires a structural
+		// change to the gles.api file.
```
Comment (dsrbecky) [inline gapis/gfxapi/gles/replay.go]: ...nevermind :-D

File: gapis/gfxapi/gles/replay.go [hunk 3]
```diff
@@ -180,6 +180,9 @@ func (a api) Replay(
 	// Cleanup
 	transforms.Add(&destroyResourcesAtEOS{})
```
Comment (dsrbecky) [inline gapis/gfxapi/gles/replay.go]: Silly question, why do we need the destroy resources in the first place?

File: gapii/cc/.gitignore [hunk 4]
```diff
@@ -1,6 +1,3 @@
-/core_imports.h
-/core_types.cpp
-/core_types.h
```
Comment (dsrbecky) [inline gapii/cc/.gitignore]: yay

File: gapii/cc/.gitignore [hunk 5]
```diff
@@ -16,5 +16,5 @@
 /vulkan_VkPhysicalDevice_exports.cpp
 /vulkan_VkQueue_exports.cpp
 /*_spy.h
-/*_spy_[0-9].cpp
+/*_spy_[0-3].cpp
```
Comment (dsrbecky) [inline gapii/cc/.gitignore]: Curiosity - why does it matter?

File: gapil/api.go [hunk 6]
```diff
@@ -189,41 +192,8 @@ func (p *Processor) resolve(base file.Path) (*semantic.API, parse.ErrorList) {
 	for i, name := range names {
 		list[i] = includes[name]
 	}
-	// Resolve all the named imports
```
Comment (dsrbecky) [inline gapil/api.go]: Another yay for deleting code

File: gapis/gfxapi/gles/replay.go (hunk 3 above)
Comment (ben-clayton) [inline gapis/gfxapi/gles/replay.go]: IIRC there was a leak on some driver that required a device to be restarted after a large number of replays.

File: gapii/cc/.gitignore (hunk 5 above)
Comment (ben-clayton) [inline gapii/cc/.gitignore]: I had a bunch'o'files knocking around that I kept on thinking were being generated and used. They weren't, they were left over from when you reduced the number of generated files.

**Comment by dsrbecky** [comment]
Could we make the colouring to be applied only when "All contexts" is selected?

**Pull request merged**